external-crates/move/move-compiler/src/shared/mod.rs

```diff
@@ -692,7 +693,7 @@ impl Default for PackageConfig {
     fn default() -> Self {
         Self {
             is_dependency: false,
-            warning_filter: WarningFilters::new(),
+            warning_filter: WarningFilters::new_for_source(),
```

Comment:
Might be worth commenting that this gets swapped for `new_for_dependency` during expansion

Reply:
It isn't though... The `new_for_dependency` function is used to initialize `Context.all_filter_alls` which in turn is used when adding new warning scopes. I am not sure what kind of comment in this place would make it more clear, but if you have a suggestion, I am happy to add it!